integr8ly/integreatly-operator
# Intly 2314 uninstall

- Add owner refs to all created objects so they are deleted when the installation is deleted.
- Pass a context through from the controller, so it can cancel the context when the installation is deleted and prevent any executing reconcilers from recreating the deleted items.
- Update unit tests
- switch to ownerutil package for owner ref management.

## Verification
- Delete any existing integreatly operatorsources
- Create operator-source pointing at `philbrookes`
- Create test namespace
- Deploy integreatly-operator to that namespace from operatorHub
- Create installation CR
- Wait for installation to finish succesfully
- Delete the CR
- Wait and see all created objects are removed
- Create installation CR again
- See installation succeed again.
